meshtastic/interfaces/ble/connection.py

```
@@ -359,6 +491,137 @@ def __init__(
+            legacy()
+            return
+        self.validator.validate_connection_request()
+
+    def _state_current_state(self) -> ConnectionState:
+        """Read current state with fallback for legacy/mocked state managers."""
+        if isinstance(self.state_manager, BLEStateManager):
+            return self.state_manager.current_state
+        legacy = getattr(self.state_manager, "_current_state", None)
+        if isinstance(legacy, ConnectionState):
+            return legacy
+        return cast(ConnectionState, getattr(self.state_manager, "current_state"))
```

Ruff (0.15.5)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)